### PR TITLE
Proper Error Handling for Build Numbers of Downloaded Sources (Non-Git repositories).

### LINKED ISSUE #9834: pdf.js version not set correctly for builds of downloaded sources (non-git repositories)
I finally (I hope) figured out why the version number of pdf.js Chrome extensions built by me didn't have the correct version.
I downloaded the 2.0.550 archive from the releases pages, set up everything as should (npm, gulp, etc) and then tried to build a Chrome extension.
The build is successfull, but the resulting extension has 2.0.0 version.
The problem is that gulpfile.js assumes it's in a git repository to determine the build number (see line 426 and on), but it's not in this case (git gives fatal: Not a git repository (or any of the parent directories): .git)
This wouldn't happen if the directory were obtained using git clone ... in which case it would be a proper git repository.

### PR DESCRIPTION
**Pull request by RonLek**
Implemented proper Error Handling along with error message for pdf.js downloaded from other sources like the PDF.JS Mozilla Release Page.
https://github.com/mozilla/pdf.js/releases

Closes #9834

### CODE REVIEW

**Review by timvandermeij — CHANGES_REQUESTED** [review]

**Comment by RonLek** [comment]
Thanks for the review @timvandermeij . I've made the requested changes. Please review.

**Comment by RonLek** [comment]
Thanks for reviewing @timvandermeij . I've made the changes.

**Comment by Snuffleupagus** [comment]
> I've made the changes.

The patch now contains mostly unrelated/unwanted changes; why did you change the indentation of the entire file? All indentation changes need to be reverted, such that the entire diff contains *nothing* more than the following changes (based on the review comments above):
```diff
diff --git a/gulpfile.js b/gulpfile.js
index c7e66a20..919201ee 100644
--- a/gulpfile.js
+++ b/gulpfile.js
@@ -449,6 +449,8 @@ gulp.task('buildnumber', function (done) {
     if (!err) {
       // Build number is the number of commits since base version
       buildNumber = stdout ? stdout.match(/\n/g).length : 0;
+    } else {
+      console.log('This is not a Git repository; using default build number.');
     }

     console.log('Extension build number: ' + buildNumber);
```

Finally, please see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

File: gulpfile.js [hunk 1]
```diff
@@ -449,6 +449,9 @@ gulp.task('buildnumber', function (done) {
     if (!err) {
       // Build number is the number of commits since base version
       buildNumber = stdout ? stdout.match(/\n/g).length : 0;
+    } else {
+      console.log('This is not a Git repository.');
+      console.log('Using default build number.');
```
Comment (Snuffleupagus) [inline gulpfile.js]: This should fit on *one* line, i.e. format it as `console.log('This is not a Git repository; using default build number.');` which is what https://github.com/mozilla/pdf.js/pull/9852#discussion_r198991482 asked for.

You also need to squash the commits, see https://github.com/mozilla/pdf.js/wiki/Squashing-Commits.

**Comment by RonLek** [comment]
Will be making a new pull request.

**Comment by RonLek** [comment]
@Snuffleupagus I'm new to contributing to GitHub. 
When I downloaded my forked PDFjs repository from GitHub and performed git log it gives me "Your current branch has no commits yet" even though I have 5 commits in my GitHub forked repository.
Can you help me?

**Comment by RonLek** [comment]
@Snuffleupagus @timvandermeij After spending a lot of time on GitHub Help pages I finally I figured out what I had to do! Can you guys please review. Sorry for the trouble.

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Looks good now. Thank you for your contribution!

**Pull request merged**